### ISSUE TITLE
Add REMUX as quality type

### ISSUE THREAD

**Issue by d2dyno** [Sonarr/Sonarr#1774]
Add REMUX (1080p) as a quality type. Since it's different in both size and spec then an x264 Encode 'BluRay-1080p' quality type, that is unusable to find remuxes. Preferred words are good, but this is a separate spec and release type, so it should have it's own quality profile.

This was already done on Radarr, a fork of Sonarr, so it should be quite simple to implement here. PR for it on Radarr: https://github.com/Radarr/Radarr/pull/900

**Comment by markus101** [comment]
> Preferred words are good, but this is a separate spec and release type, so it should have it's own quality profile.

It's still a bluray source, not unlike x264 and x265, which also have a size difference.

I don't think either @Taloth or I think this makes sense as a new quality, so unless there is a compelling reason why it should be it's own quality, I don't see it happening.

**Comment by d2dyno** [comment]
The _source_ is the same, but the output is very different. As I'm sure you know, it's uncompressed (lossless) vs. compressed (lossy). One is untouched, one is run through a program by an encoder, with whichever settings they deem fit. I'm shocked you won't see the difference.

With many popular release groups doing REMUXes as their standard, while also having the highest snatch count above encodes, this will be a sought out feature.

If the code is already here, (relatively) easy to integrate, I don't see the reason to shoot this down so quickly.

**Comment by Taloth** [comment]
Actually, most REMUX are just the original h264/vc-1 stream. True uncompressed streams pretty much never get posted and thus remuxes aren't lossless but always lossy, they are just high quality.
btw "Seeing the difference" is subjective, besides markus made no claims in that regard.

Point is that we have no intention to add quality definitions for every single one out there. Atm there are 3 sources (tv, web, disk) and 4 resolutions, Sonarr doesn't care about the subdivisions therein. The notable exception being Raw-HD = raw mpeg2 .ts files.
Until we have some kind of grouping to keep it simple for casual users we will not add additional ones.

**Comment by LSL1337** [comment]
yeah but it's a different quality.
it's like your mother would say 720p and 1080p is the same, it's both from bluray. If you want one and not the other, currently you can't distinguish between them with sonarr, maybe with global tags, but that's a very crude way. I think there is a HUGE disconnect between public and private tracker users. for some people there is only 1 result for bluray, but for others there are a lot, and it's still the same for sonarr. good luck automating that, than you are back to manual choosing, which is almost the same as not using sonarr.
custom quality would solve this problem. (essentially allow us to use our own custom regex-es, and rank them in profiles as we see it fit. Make it an advanced setting, so it doesn't confuse other people. More choice/customization, the better. (it would solve, audio, codec, release group, any other tag related problem!)

just my 2cents

**Comment by d2dyno** [comment]
80% of remuxes are AVC. At least the ones from 2015+

I agree with @LSL1337, there must be a disconnect. Private HD trackers are almost Remux oriented. Public are hardly so. I'm a part of over 15 HD private trackers (using Jackett).

**Comment by Taloth** [comment]
As I said, we will not be adding a separate remux quality until we can properly support the grouping of qualities to keep things simple for the average user. Sonarr is intended for the average user and does not aim to satisfy every use-case, so anything we do is a trade-off. Sonarr isn't designed to be flexget.

I think we can all agree that remuxes are higher quality releases than your average re-encode, that wasn't disputed. Remuxes have similarities to Raw-HD, in that it potentially deserves a class of it's own, but it's not gonna get it now.

On a side note, I suspect we have a different definition of 'lossless'. But it doesn't have bearing on this discussion, coz that wasn't about how lossless the source is.

**Comment by LSL1337** [comment]
"until we can properly support the grouping of qualities to keep things simple for the average user"
Can you be a little specific, what changes are you planing?
I head something about dumping quality profiles, and using tags instead.

**Comment by Taloth** [comment]
Grouping first got in the picture when we started to think about webrip vs web-dl (which is a problem for a much larger subset of our userbase), some ppl don't give a shit about whether they get dl or rip, and don't want one to be considered better than the other, but other ppl do have preferences. We've been thinking of a way to do both and at the same time figure out how the heck we're gonna realize 'preferred keywords'. 
It gets complicated fast, and we can't have that for Mr. & Mrs. Smith.

**Comment by LSL1337** [comment]
Thanks for the answer, but i'm still not clear what you mean by grouping. right now webdl and webrip is in the same group (webdl). do you want to seperate that? or not, cos it's difficult?

I don't see how is it confusing or difficult to implement what i proposed earlier.
the current system is fine for the avarage user imho. no need to change it.
you make and advanced only setting, where people can edit the tags (regex, string, whatever you wanna call it) which put releases into quality profiles. they can do everything with that system basicly, and can rank them in whatever order they want in the profiles they have created. a quality could be called amzn, or webdl, or internal or -myfavourite group.  instad of the current hdtv-1080p and webdl-1080p.

if an avarage joe wants to go deeper, a 10 line forum guide can explain the system, and how can he edit it, if he chooses.
if this is difficult for somebody, i don't think they could get sonarr up and running by themselves, but it's just my 2 cents.

**Comment by markus101** [comment]
> Thanks for the answer, but i'm still not clear what you mean by grouping. right now webdl and webrip is in the same group (webdl). do you want to seperate that? or not, cos it's difficult?

Correct, right now they are in the same group, we want to split them out, but allow them to be in a profile with the same weight, so Sonarr could grab WEB-DL or WEB-Rip, but not both and also upgrade to a higher quality (this is covered by another issue you commented on, https://github.com/Sonarr/Sonarr/issues/1781 where I have added an example).

 > I don't see how is it confusing or difficult to implement what i proposed earlier.
the current system is fine for the avarage user imho. no need to change it.
you make and advanced only setting, where people can edit the tags (regex, string, whatever you wanna call it) which put releases into quality profiles. they can do everything with that system basicly, and can rank them in whatever order they want in the profiles they have created. a quality could be called amzn, or webdl, or internal or -myfavourite group. instad of the current hdtv-1080p and webdl-1080p.

TLDR no.

This would is exactly what we don't want to do, this is what FlexGet is/can be. For a small group of users this may unlock some potential, but for everyone else it would be completely unnecessary and come with a 100% chance of being used wrong and having to be addressed via support. Unless there is a skill testing question to unlock this mode I'm sure we'd get questioned on how to build the regex and test it and on and on, support hell. Most users aren't developers, we don't want to expose them to things they don't need to see. Re-ordering qualities within a profile trips people up enough already (not a lot, but enough) and this wouldn't help any.

If the qualities are there that you want, what does a complicated custom quality setup give you?

> if an avarage joe wants to go deeper, a 10 line forum guide can explain the system, and how can he edit it, if he chooses.

The average joe shouldn't be ignored because a fraction of users would use a complicated feature. Like @Taloth said, we're not going to customize everything to make FlexGet, if you want that, use FlexGet.

> if this is difficult for somebody, i don't think they could get sonarr up and running by themselves, but it's just my 2 cents.

It's not really hard to double click an icon and setup basic settings, it becomes a lot more complicated when you suddenly see something way out of your realm and don't understand it. Not to mention the people that had a friend set it up and then they tweak things, you shouldn't need to be a developer, be a rocket scientist or read a bunch of documentation to understand how to use Sonarr.

**Comment by LSL1337** [comment]
again, thanks for the detailed answer.
this grouping is a different ballgame, but it would help the webdl-webrip difference if you are indifferent about them certanly.

do you guys have any plans on how to implement something for, or still in the brainstorming phase?

btw telling me to use flexget is just nonsense, radarr has so many extra functions, that a customized string wouldn't make it 'like flexget'.

still, cheers

**Comment by markus101** [comment]
> do you guys have any plans on how to implement something for, or still in the brainstorming phase?

We haven't gone into much details on it, mostly just brainstorming. It's a pretty significant overhaul though and has UX/UI considerations to investigate.

> btw telling me to use flexget is just nonsense, radarr has so many extra functions, that a customized string wouldn't make it 'like flexget'.

*Sonarr
If you want to control "qualities" at that level that's the option (AFAIK at least), but like you said, a trade off.

**Comment by d2dyno** [comment]
In my mind, if something lacks in utter simplicity (one click install for example), it can be made up for in documentation. I find it hard to believe the user isn't expected to read any documentation, that's ignorance on the user's part. Isn't it an unwritten (but often written) rule to read instructions before asking for help? Anyways, that doesn't help much I imagine. Just discussion.

Everyone I've come across who uses Sonarr, is an enthusiast who understands the system of torrents pretty well, but of course I don't have the historical data that you do. Right now, **_I_ don't even understand** what Raw-HD is supposed to represent, as there are no releases tagged as such. Changing that value to REMUX would make it less ambiguous for both simple and advanced users.

**Comment by markus101** [comment]
>  I find it hard to believe the user isn't expected to read any documentation

A difference between none and having to read a lot though, plus if I've learned anything it's that people don't read. We have tooltips in the UI with warnings and help text and get questions about it, best case we point it out and they realize their mistake or worst case we point it out, they don't understand it and need more help, perhaps some of that is on use for being unclear or using domain specific terms, but largely it seems people don't read.

> I find it hard to believe the user isn't expected to read any documentation, that's ignorance on the user's part. Isn't it an unwritten (but often written) rule to read instructions before asking for help?

I'd hope so, but even still people want validation or confirmation and the more complicated the feature the more likely people will ask questions.

> Everyone I've come across who uses Sonarr, is an enthusiast who understands the system of torrents pretty well, but of course I don't have the historical data that you do.

A lot do, but it's a mixed bag, some people have barely touched linux and suddenly want Sonarr on it or getting into usenet or torrents for the first time.

> Right now, I don't even understand what Raw-HD is supposed to represent, as there are no releases tagged as such.

1080i compressed, not wildly popular, but a legitimate quality.

> Changing that value to REMUX would make it less ambiguous for both simple and advanced users.

That would break anyone wanting to use `RAW-HD` and wouldn't help with the difference in 2160p vs 1080p.

Anyways I think we're getting off base here, we have plans to add Remux as separate quality (well two qualities, 1080p and 2160p), but we want to adding grouping first.

**Comment by d2dyno** [comment]
> 1080i compressed, not wildly popular, but a legitimate quality.
> That would break anyone wanting to use RAW-HD and wouldn't help with the difference in 2160p vs 1080p.

Ah, once described, I understand now. Yes, that would be undesirable.
> we have plans to add Remux as separate quality (well two qualities, 1080p and 2160p), but we want to adding grouping first.

Yay! *Evil laugh ensues, as I've broken your will. muahahaha*

**Comment by Ninjouz** [comment]
Hey guys, personally I'd really love to set up my custom quality types.
Like I'd like to set a Amazon WEBRIP (much better quality overall + integrated subs) over HDTV releases over WEB-DL.

I really think that being able to setup custom qualities (giving us the possibility to setup our own regexp) would simplify your job to meet everyone's expectations (ofc with reordering still possible).

Hope you can consider this request :)

Thanks for your awesome work !

**Comment by markus101** [comment]
Separating WEBRIPs is covered by: https://github.com/Sonarr/Sonarr/issues/1781 We have no plans to implement and support custom qualities.

**Comment by piejanssens** [comment]
`Series S01E06 Episode Title 2160p DTS-HD MA 5 1 HEVC REMUX-MEDIUMRARE` is currently imported as HDTV-2160p...

**Comment by eversins** [comment]
When is v3.x planned? I am waiting on most series until remux is implemented.

**Comment by corykim** [comment]
The PR at the top of this request is *such* a low-risk change, will benefit users, and will not really make a future redesign any more complicated. Are there users telling you they *don't* want it?

**Comment by markus101** [comment]
This will be part of v3 (whenever it releases, no ETA).
